utils/dom.js

```diff
@@ -45,7 +75,9 @@ export function isHorizontalEdge( container, isReverse, collapseRanges = false )
 	let range = selection.rangeCount ? selection.getRangeAt( 0 ) : null;
 	if ( collapseRanges ) {
 		range = range.cloneRange();
-		range.collapse( isReverse );
+
+		// Collapse to the extent of the selection by direction.
+		range.collapse( ! isSelectionForward( selection ) );
```

Comment:
Is the goal to set this range to the document? This is just used for calculation purposes (a clone) right?

Reply:
I'm not quite sure here how the direction of a selection relates to this check. This would collapse a range at the edge of a field to the end or the start depending on direction (so in one case this will collapse to the end and the check fails), while we should check whether or not the range touches the edge? See https://github.com/WordPress/gutenberg/commit/1bb965830820a23ec5ab6768a432c01315a7e832